odoo/tools/cloc.py

```diff
@@ -175,6 +195,44 @@ def count_customization(self, env):
         for f in env['ir.model.fields'].browse(data.keys()):
             self.book(data[f.id] or "odoo/studio", "ir.model.fields/%s: %s" % (f.id, f.name), self.parse_py(f.compute))
 
+        if not imported_module:
+            return
+
+        # Count qweb view only from imported module
+        query = """
+            SELECT view.id, data.module
+              FROM ir_ui_view view
+        INNER JOIN ir_model_data data ON view.id = data.res_id AND data.model = 'ir.ui.view'
+        INNER JOIN ir_module_module mod ON mod.name = data.module AND mod.imported = True
+             WHERE type = 'qweb'
+        """
+        env.cr.execute(query)
+        custom_views = {r[0]: r[1] for r in env.cr.fetchall()}
+        for view in env['ir.ui.view'].browse(custom_views.keys()):
+            modue_name = custom_views[view.id]
+            self.book(modue_name, "/%s/views/%s.xml" % (modue_name, view.name), self.parse_xml(view.arch_base))
+
+        # Count js, xml, css/scss file from imported module
+        query = r"""
+            SELECT attach.id, data.module
+              FROM ir_attachment attach
+        INNER JOIN ir_model_data data ON attach.id = data.res_id AND data.model = 'ir.attachment'
+        INNER JOIN ir_module_module mod ON mod.name = data.module AND mod.imported = True
+             WHERE attach.name ~ '.*\.(js|xml|css|scss)$'
+        """
+        env.cr.execute(query)
+        uploaded_file = {r[0]: r[1] for r in env.cr.fetchall()}
+        for attach in env['ir.attachment'].browse(uploaded_file.keys()):
+            module_name = uploaded_file[attach.id]
+            ext = os.path.splitext(attach.url)[1].lower()
+            if ext in VALID_EXTENSION:
+                if len(attach.datas) > MAX_FILE_SIZE:
+                    self.book(module_name, attach.url, (-1, "Max file size exceeded"))
+                    continue
+
+                content = base64.b64decode(attach.datas).decode('latin1')
```

Comment:
Latin1?
What about using the `.raw` attribute?
```suggestion
                content = attach.raw
```

Reply:
I didn't know raw, it's a good idea 
but still the parse method use string and not binary so it needs to be decoded.
Antony choose latin1, because decoding in latin1 never raise a exception (in contrary of decode('utf-8')) and the char that we take care for counting line are the same in utf-8 and latin1

```suggestion
                content = attach.raw.decode('latin1')
```

Reply:
It should be a comment in the code, then. Because it's not obvious why it is done this way.